src/Adapter/CustomerService/QueryHandler/GetCustomerThreadForViewingHandler.php

```diff
@@ -0,0 +1,386 @@
+use PrestaShop\PrestaShop\Core\Domain\CustomerService\Query\GetCustomerThreadForViewing;
+use PrestaShop\PrestaShop\Core\Domain\CustomerService\QueryHandler\GetCustomerThreadForViewingHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\CustomerService\QueryResult\CustomerInformation;
+use PrestaShop\PrestaShop\Core\Domain\CustomerService\QueryResult\CustomerThreadMessage;
+use PrestaShop\PrestaShop\Core\Domain\CustomerService\QueryResult\CustomerThreadTimeline;
+use PrestaShop\PrestaShop\Core\Domain\CustomerService\QueryResult\CustomerThreadTimelineItem;
+use PrestaShop\PrestaShop\Core\Domain\CustomerService\QueryResult\CustomerThreadView;
+use PrestaShop\PrestaShop\Core\Domain\CustomerService\ValueObject\CustomerThreadId;
+use PrestaShop\PrestaShop\Core\Domain\CustomerService\ValueObject\CustomerThreadStatus;
+use PrestaShop\PrestaShop\Core\Domain\Language\ValueObject\LanguageId;
+use Product;
+use Symfony\Component\Translation\TranslatorInterface;
+use Tools;
+use Validate;
+
+/**
+ * @internal
+ */
+final class GetCustomerThreadForViewingHandler implements GetCustomerThreadForViewingHandlerInterface
+{
+    /**
+     * @var Context
+     */
+    private $context;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @param Context $context
+     */
+    public function __construct(Context $context)
+    {
+        $this->context = $context;
+        $this->translator = $context->getTranslator();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(GetCustomerThreadForViewing $query)
+    {
+        $customerThread = $this->getCustomerThread($query->getCustomerThreadId());
+
+        $messages = CustomerThread::getMessageCustomerThreads($query->getCustomerThreadId()->getValue());
+
+        return new CustomerThreadView(
+            $query->getCustomerThreadId(),
+            new LanguageId((int) $customerThread->id_lang),
+            $this->getAvailableActions($customerThread),
+            $this->getCustomerInformation($customerThread),
+            $this->getContactName($customerThread),
+            $this->getCustomerThreadMessages($messages),
+            $this->getTimeline($messages, $customerThread)
+        );
+    }
+
+    /**
+     * @param array $messages
+     *
+     * @return CustomerThreadMessage[]
+     */
+    private function getCustomerThreadMessages(array $messages)
+    {
+        $threadMessages = [];
+
+        foreach ($messages as $key => $message) {
+            $employeeImage = null;
+
+            if ($message['id_employee']) {
+                $employee = new Employee($message['id_employee']);
+                $employeeImage = $employee->getImage();
+            }
+
+            $attachmentFile = null;
+
+            if (isset($message['file_name']) && $message['file_name'] != '') {
+                $attachmentFile = _THEME_PROD_PIC_DIR_ . $message['file_name'];
```

Comment:
If this is a filepath, can we add a `file_exists()` here to make sure the attachment actually exists ?